common/json.c

```diff
@@ -216,3 +216,54 @@ jsmntok_t *json_parse_input(const char *input, int len, bool *valid)
 
 	return toks;
 }
+
+void json_tok_print(const char *buffer, const jsmntok_t *params)
+{
+	const jsmntok_t *first = params;
+	const jsmntok_t *last = json_next(params);
+	printf("size: %d, count: %ld\n", params->size, last - first);
+	while (first != last) {
+		printf("%ld. %.*s, %d\n", first - params,
+		        first->end - first->start, buffer + first->start,
+			first->type);
+		first++;
+	}
+	printf("\n");
+}
+
+jsmntok_t *json_tok_copy(const tal_t *ctx, const jsmntok_t *tok)
+{
+	const jsmntok_t *first = tok;
+	const jsmntok_t *last = json_next(tok);
+
+	if (!tok)
+		return NULL;
+	jsmntok_t *arr = tal_arr(ctx, jsmntok_t, last - first);
+	jsmntok_t *dest = arr;
+	while (first != last)
+		*dest++ = *first++;
+
+	return arr;
+}
+
+void json_tok_remove(jsmntok_t **tokens, jsmntok_t *tok, size_t num)
+{
+#if DEVELOPER
+	assert(*tokens);
+	assert((*tokens)->type == JSMN_ARRAY || (*tokens)->type == JSMN_OBJECT);
+#endif
```

Comment:
i would assert these unconditionally, TBH.  We're in for worse (and more confusing!) hurt if they're not true...